### PR TITLE
feat: implement authorizations, write tests and update readme

### PR DESCRIPTION
**Pull request by sihamais**
### Changes proposed in this pull request
-  implement authorizations
- write tests
- update readme
- update shop and promotion creation to extract shop id from logged in user

[FID-125]

[FID-125]: https://msc-etna.atlassian.net/browse/FID-125?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/factory.ts [hunk 1]
```diff
@@ -78,6 +83,44 @@
...
+    return request(this.app.getHttpServer())
+      .put(path)
+      .set('Authorization', 'bearer some-token')
+      .set('Accept', 'application/json');
+  }
+
+  public async delete(path: string): Promise<request.Test> {
+    return request(this.app.getHttpServer())
+      .delete(path)
+      .set('Authorization', 'bearer some-token')
```
[the remark this answers is not in the document]
Comment (sihamais) [reply test/factory.ts:110]: Used in tests

File: test/factory.ts [hunk 2]
```diff
@@ -78,6 +83,44 @@
...
+    return request(this.app.getHttpServer())
+      .post(path)
+      .set('Authorization', 'bearer some-token')
+      .set('Accept', 'application/json');
+  }
+
+  public put(path: string): request.Test {
+    return request(this.app.getHttpServer())
+      .put(path)
+      .set('Authorization', 'bearer some-token')
```
[the remark this answers is not in the document]
Comment (sihamais) [reply test/factory.ts:103]: Used in tests

File: test/factory.ts [hunk 3]
```diff
@@ -78,6 +83,44 @@
     await this._app.close();
   }
 
+  public async get(path: string): Promise<request.Test> {
+    return request(this.app.getHttpServer())
+      .get(path)
+      .set('Authorization', 'bearer some-token')
```
[the remark this answers is not in the document]
Comment (sihamais) [reply test/factory.ts:89]: Used in tests

**Review by JeremCy — APPROVED** [review]
LGTM

**Pull request merged**